javagen/src/main/java/com/azure/autorest/mapper/ProxyParameterMapper.java

```diff
@@ -62,11 +63,12 @@ public ProxyMethodParameter map(Parameter parameter) {
         builder.rawType(wireType);
 
         IType clientType = wireType.getClientType();
-        if (settings.isDataPlaneClient() && !(clientType instanceof PrimitiveType)) {
+
+        if (settings.isDataPlaneClient()) {
             if (parameterRequestLocation == RequestParameterLocation.BODY /*&& parameterRequestLocation != RequestParameterLocation.FormData*/) {
                 clientType = ClassType.BinaryData;
-            } else {
-                clientType = ClassType.String;
+            } else if(!(clientType instanceof PrimitiveType)){
+                clientType = SchemaUtil.mapDataPlaneNonPrimitiveParameterType(clientType);
             }
```

Comment:
Why not put the `body` part into the utility method as well?

Just call the method `removeModelFromParameter`